src/js/sdk.tsx

```diff
@@ -65,17 +65,25 @@ export function init(passedOptions: ReactNativeOptions): void {
     // eslint-disable-next-line deprecation/deprecation
     ?? passedOptions.transportOptions?.bufferSize
     ?? DEFAULT_OPTIONS.maxQueueSize;
+
+  let enableNative = passedOptions.enableNative !== undefined
+    ? passedOptions.enableNative
+    : DEFAULT_OPTIONS.enableNative;
+  // If custom transport factory fails the SDK won't initialize
+  let chosenTransport: ((transportOptions: ReactNativeTransportOptions) => Transport) | null
+    = passedOptions.transport
+    || makeNativeTransportFactory({
+      enableNative: enableNative
+    });
+  if (chosenTransport == null) {
```

Comment:
M: Could we make the decision directly based on `NATIVE.isNativeAvailable` so even if people supply their own transport the native is disabled if not available?

Reply:
I implemented that, since this only affects the value of EnableNative, I rolled back the original logic state since we no longer need the altered condition to check if enableNative should be set to true or false.